airflow/api/common/experimental/__init__.py

```diff
@@ -45,3 +45,10 @@ def check_and_get_dagrun(dag: DagModel, execution_date: datetime) -> DagRun:
                          .format(execution_date, dag.dag_id))
         raise DagRunNotFound(error_message)
     return dagrun
+
+
+def check_and_get_dags():
+    """Get the dag ids present in the dagbag"""
+    dagbag = DagBag()
```

Comment:
Calling `DagBag()` is a quite expensive action. You may want to get the list of dag ids from DB instead.

On the other hand, my personal preference is not to have a new separate function if what you're doing inside is very straightforward & not being used at multiple locations (here this `check_and_get_dags()` function is only used at one location).

Reply:
You're right - using the DB session instead of `DagBag()` significantly reduced request time. I have removed the `check_and_get_dags()` function as well.